### PR TITLE
[Feature] Formulas for http performance test: http-load & http-config

### PR DESCRIPTION
**Pull request by douglasvinter**
# Pull Request

## What I did
- Created a new command super set named `http` which is suppose to provide http testing & development tools
- Created the very first http helper command named `rit http perform load-test`, which is an simple HTTP performance testing tool
- Created the second http command, named `rit http generate http-config`, which generates the target http host into a json file on users `~/.rit` folder

## How I did it
- Used some HTTP tools as inspiration (go lang vegeta, JMeter, Gatlin) to create the very first, simple and light weight ritchie's performance tool
- Used go routines without a mutex to increment correlation-ID and sequency-ID on HTTP Headers (That might requires security off in some target applications)
- The concurrency model (a.k.a threading spawn) is based on Brian Goetz thread pool formula

## How to verify it

- build the formula, generate the default config file using the command `rit http perform load-test`, edit the created file (if you want to) and then run the command `rit http perform load-test`
- the formula will ask for the test duration (in seconds) and the desired max threads (go routines) to be used, if you pass the value 0, the defaultMaxWorkers value will be used
- Dots ("..") will be printed while running as a feedback that its running, and will notify you when its done with the results file full path for you to copy and paste, which is by default named: `ritman-test-result-<timestamp>.json`

## What kind of change does this PR make

- [ ] Major
- [X] Minor
- [ ] Patch

## Description for the changelog
- Added a formula for HTTP performance test
- Added a formula to generate the base configuration template

### CODE REVIEW

File: http/generate/http-config/README.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+   }
+```
+2. You can easily use all the necessary HTTP Verbs for testing on the "method" field: GET, PUT, POST, PATCH, HEAD, OPTIONS, DELETE
+3. you can also add an authorization token for private API testing in the "headers" field:
+```json
+    {
+        "Authorization": "<Bearer jwt> or <basic> or wathever token that fits your Authorization auth header"
+    }
+```
+4. The "body" field supports any valid JSON format, so you can fullfill with your disired test content.
```
Comment (GuillaumeFalourd) [inline http/generate/http-config/README.md:32]: Great description of the formula 👍🏼  
Could you just update it to look more like the default README we recommend to use please @douglasvinter ?

Example: https://github.com/ZupIT/ritchie-formulas-demo/tree/master/demo/hello-world

File: http/perform/load-test/README.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+}
+```
+Note:
+- The "histogram" field is a series of result where the key is the second that the request was fired to the target endpoint.
+- You may notice some "-1" in the "statusCode" field, which means the requests either took a Socket Timeout (maximum wait time is 60 seconds) or your Operation System may exceeded the number of Created Socket in the Socket Stack (65k approx). So make sure to NOT RUN the http service in the same machine you'll run the load testing.
+
+Acronyns:
+    - avg - stands for average
+    - rps - stands for rate per seconds
+    - ms - stands for milliseconds
```
Comment (GuillaumeFalourd) [inline http/perform/load-test/README.md:132]: Great description of the formula 👍🏼  
Could you just update it to look more like the default README we recommend to use please @douglasvinter ?

Example: https://github.com/ZupIT/ritchie-formulas-demo/tree/master/demo/hello-world

**Review by GuillaumeFalourd — CHANGES_REQUESTED** [review]
Just a couple of comments regarding the README files, the rest looks good to me 👍🏼

File: http/generate/http-config/README.md (hunk 1 above)
Comment (douglasvinter) [reply http/generate/http-config/README.md]: Done

**Review by GuillaumeFalourd — APPROVED** [review]

**Comment by GuillaumeFalourd** [comment]
LGTM @douglasvinter 🚀 
Just sign-off the DCO please, for the PR to be approved 👍🏼

**Comment by douglasvinter** [comment]
DCO done

**Review by GuillaumeFalourd — APPROVED** [review]

**Comment by GuillaumeFalourd** [comment]
Thank you for contributing @douglasvinter , with one more accepted PR you'll be able to participate the [ZupIT challenge for the hacktoberfest](https://insights.zup.com.br/hacktoberfest)! 🚀

**Pull request merged**